src/clients/BundleDataClient.ts

```diff
@@ -215,83 +238,162 @@ export class BundleDataClient {
-      bundleEvaluationBlockRanges,
-      this.spokePoolClients,
-      logData,
-      attemptToLoadFromArweave
-    );
-    const combinedRefunds = getRefundsFromBundle(bundleFillsV3, expiredDepositsToRefundV3);
+    let combinedRefunds: CombinedRefunds;
+    const arweaveData = await this.loadArweaveData(bundleEvaluationBlockRanges);
+    if (arweaveData === undefined) {
+      combinedRefunds = this.getApproximateRefundsForBlockRange(
+        chainIds,
+        bundleEvaluationBlockRanges,
+        filteredRefundAddresses
+      );
+    } else {
+      const { bundleFillsV3, expiredDepositsToRefundV3 } = arweaveData;
+      combinedRefunds = getRefundsFromBundle(bundleFillsV3, expiredDepositsToRefundV3);
+    }
 
     // The latest proposed bundle's refund leaves might have already been partially or entirely executed.
     // We have to deduct the executed amounts from the total refund amounts.
     return this.deductExecutedRefunds(combinedRefunds, bundle);
   }
 
+  // @dev This helper function should probably be moved to the InventoryClient
+  getApproximateRefundsForBlockRange(
+    chainIds: number[],
+    blockRanges: number[][],
+    filteredRefundAddresses: string[]
+  ): CombinedRefunds {
+    const refundsForChain: CombinedRefunds = {};
+    for (const chainId of chainIds) {
+      if (this.spokePoolClients[chainId] === undefined) {
+        continue;
+      }
+      const chainIndex = chainIds.indexOf(chainId);
+      this.spokePoolClients[chainId]
+        .getFills()
+        .filter(
+          (fill) =>
+            filteredRefundAddresses.includes(fill.relayer) &&
+            fill.blockNumber >= blockRanges[chainIndex][0] &&
+            fill.blockNumber <= blockRanges[chainIndex][1]
+        )
+        .forEach((fill) => {
+          const { chainToSendRefundTo, repaymentToken } = getRefundInformationFromFill(
+            fill,
+            this.clients.hubPoolClient,
+            blockRanges,
+            chainIds
+          );
+          // Assume that lp fees are 0 for the sake of speed. In the future we could batch compute
+          // these or make hardcoded assumptions based on the origin-repayment chain direction. This might result
+          // in slight over estimations of refunds, but its not clear whether underestimating or overestimating is
+          // worst from the relayer's perspective.
+          const refundAmount = fill.inputAmount;
+          refundsForChain[chainToSendRefundTo] ??= {};
+          refundsForChain[chainToSendRefundTo][repaymentToken] ??= {};
+          if (refundsForChain[chainToSendRefundTo][repaymentToken][fill.relayer]) {
+            refundsForChain[chainToSendRefundTo][repaymentToken][fill.relayer] =
+              refundsForChain[chainToSendRefundTo][repaymentToken][fill.relayer].add(refundAmount);
+          } else {
+            refundsForChain[chainToSendRefundTo][repaymentToken][fill.relayer] = refundAmount;
+          }
+        });
```

Comment:
Small simplification that deletes a few lines:

```suggestion
          const { relayer, inputAmount: refundAmount } = fill;
          refundsForChain[chainToSendRefundTo] ??= {};
          refundsForChain[chainToSendRefundTo][repaymentToken] ??= {};
          const existingRefundAmount = refundsForChain[chainToSendRefundTo][repaymentToken][relayer] ?? bnZero;
          refundsForChain[chainToSendRefundTo][repaymentToken][relayer] = existingRefundAmount.add(refundAmount);
```